python/cudf_polars/tests/test_drop_nulls.py

```diff
@@ -0,0 +1,27 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+
+import polars as pl
+
+from cudf_polars.testing.asserts import assert_gpu_result_equal
+
+
+@pytest.fixture(
+    params=[
+        [1, 2, 1, 3, 5, None, None],
+        [1.5, 2.5, None, 1.5, 3, float("nan"), 3],
+        [],
+        [None, None],
```

Comment:
These two tests fail for me. Looks like theres some issues constructing empty columns that I'm looking into.

Reply:
If there is no dtype, polars will create a column which has dtype `arrow::null`. We convert that to a `cudf::type_id::EMPTY` column type, but AFAICT, libcudf barfs on basically all operations on `EMPTY` columns, so I complain on construction.

Reply:
I see, I can get around this issue in this PR by making sure that the data is typed before I try dropping nulls. For a moment I wondered if this would create any UX issues - In particular I think this means that if I create a dataframe with an untyped list of nulls `[None, None]` or even an empty list `[]` it disables collecting the frame and I think any operations that materialize that column:

```python
>>> df = pl.DataFrame({'a': [None, None], 'b':[1,2]}).lazy()
>>> df.collect(post_opt_callback=partial(execute_with_cudf, raise_on_fail=True)) # error 
```

IIUC then the presence of an untyped null column in the data means we'll end up falling back to CPU for ops that would be permissible otherwise, do we care about that?

Reply:
no, because you can't do almost anything with an EMPTY column in libcudf. If this turns out to be problematic we can change it later

Reply:
What happened with this discussion? It looks like this test is passing now. Is this behavior already the default somewhere?

Reply:
We now fall back to `Float64` if a column has the empty dtype, at least for these tests.